bootstrap/pkg/kfapp/gcp/gcp.go

```diff
@@ -1732,68 +1659,104 @@ func (gcp *Gcp) Generate(resources kftypes.ResourceEnum) error {
+		if err := gcp.kfDef.SetApplicationParameter("iap-ingress", "ipName", gcp.kfDef.Spec.IpName); err != nil {
+			return errors.WithStack(err)
+		}
+		if err := gcp.kfDef.SetApplicationParameter("iap-ingress", "hostname", gcp.kfDef.Spec.Hostname); err != nil {
+			return errors.WithStack(err)
+		}
+		if err := gcp.kfDef.SetApplicationParameter("profiles", "admin", gcp.kfDef.Spec.Email); err != nil {
+			return errors.WithStack(err)
+		}
 	}
 	if *pluginSpec.CreatePipelinePersistentStorage {
+		log.Infof("Configuring pipeline, minio, and mysql applications")
 		minioPdName := gcp.kfDef.Name + "-storage-artifact-store"
 		mysqlPdName := gcp.kfDef.Name + "-storage-metadata-store"
-		gcp.kfDef.Spec.ComponentParams["pipeline"] = setNameVal(gcp.kfDef.Spec.ComponentParams["pipeline"], "mysqlPd", mysqlPdName, false)
-		gcp.kfDef.Spec.ComponentParams["pipeline"] = setNameVal(gcp.kfDef.Spec.ComponentParams["pipeline"], "minioPd", minioPdName, false)
-		if strings.HasPrefix(gcp.kfDef.Spec.PackageManager, "kustomize") {
-			gcp.kfDef.Spec.ComponentParams["minio"] = setNameVal(gcp.kfDef.Spec.ComponentParams["minio"], "minioPd", minioPdName, false)
-			gcp.kfDef.Spec.ComponentParams["mysql"] = setNameVal(gcp.kfDef.Spec.ComponentParams["mysql"], "mysqlPd", mysqlPdName, false)
+
+		// TODO(jlewi): I think we can delete this. It looks like we still had componentParams pipeline
+		// but pipeline wasn't listed in components which explains why when we migrated to Application
+		// there wasn't any Application named pipeline.
+		// Looking at https://github.com/kubeflow/manifests/tree/master/pipeline
+		// there is no base application named pipeline.
+		//if err := gcp.kfDef.SetApplicationParameter("pipeline", "mysqlPd", mysqlPdName); err != nil {
+		//	return errors.WithStack(err)
+		//}
+		//if err := gcp.kfDef.SetApplicationParameter("pipeline", "minioPd", minioPdName); err != nil {
+		//	return errors.WithStack(err)
+		//}
+		if err := gcp.kfDef.SetApplicationParameter("minio", "minioPd", minioPdName); err != nil {
+			return errors.WithStack(err)
+		}
+		if err := gcp.kfDef.SetApplicationParameter("mysql", "mysqlPd", mysqlPdName); err != nil {
+			return errors.WithStack(err)
 		}
 	}
-	gcp.kfDef.Spec.ComponentParams["notebook-controller"] = setNameVal(gcp.kfDef.Spec.ComponentParams["notebook-controller"], "injectGcpCredentials", "true", false)
 
-	for _, comp := range gcp.kfDef.Spec.Components {
-		if comp == "spartakus" {
-			rand.Seed(time.Now().UnixNano())
-			gcp.kfDef.Spec.ComponentParams["spartakus"] = setNameVal(gcp.kfDef.Spec.ComponentParams["spartakus"],
-				"usageId", strconv.Itoa(rand.Int()), true)
+	// TODO(jlewi): Why is this hard coded here?
+	if err := gcp.kfDef.SetApplicationParameter("notebook-controller", "injectGcpCredentials", "true"); err != nil {
+		return errors.WithStack(err)
+	}
+
+	// TODO(jlewi): Why are we setting this here?
```

Comment:
so that rand.Int() in usageId will be random?

Reply:
Updated the comment; I meant here in gcp.go as opposed to kustomize.go